lib/active_merchant/billing/gateways/adyen.rb

```diff
@@ -207,7 +207,11 @@ def success_from(action, response)
       def message_from(action, response)
         case action.to_s
         when 'authorise'
-          response['refusalReason'] || response['resultCode'] || response['message']
+          if response['refusalReason'] && response['additionalData'] && response['additionalData']['refusalReasonRaw']
+            "#{response['refusalReason']} | #{response['additionalData']['refusalReasonRaw']}"
+          else
+            response['refusalReason'] || response['resultCode'] || response['message']
+          end
```

Comment:
I tried a few iterations of this and gave up on making it more elegant, so suggestions are welcome.

Reply:
@curiousepic - To start off, you could eliminate the `case` statement use an if/else on the action instead instead - there's the 'authorize` action and then every other action.

Reply:
Because of the nested hash access, there isn't much else you can do right now. Once we're on Ruby 2.3, `.dig` should help here http://ruby-doc.org/core-2.3.0/Hash.html#method-i-dig

Reply:
You could take this a step further by isolating moving the nested hash logic into their own methods. Something like:
```
if action.to_s == 'authorize'
  authorization_response_from(response)
else
  other_response_from(response)
end
```

This at least makes things a bit more descriptive.

Reply:
Sounds good, thanks 👍